### DISCUSSION TITLE
Loans and Overdrafts Interest Calculations

### DISCUSSION THREAD

**Discussion by mugmuggy** [CorsixTH/CorsixTH#2525] [Research Room]
#### Describe the issue
Opening for any discussion.

For Loans, on the bank manager screen is an interest payment, calculated as loan * rate / 12.
In the original this was the amount due, if the loan was not paid off by the end of the month.

Corsix calculates on daily basis (Hospital:onEndDay), loan * rate / 365, sums these and presents these at the end of the month. It also means some months have an interest payment greater/smaller than specified on the bank manager summary compared to the transactions.

So original is slightly more forgiving here, as in paying off loan removes the interest for the entire month, but also has the opposite, and if you only take out a loan at the end of a month, hits you with much more interest than Corsix.

This is some game balance here, as you can fiddle with the figures, in the original, take out a loan for expansion, and try to pay some or all of it back before the end of month; say dealing with a large emergency and adding another room when cash is low, but emergency bonus and treatment payments cover the loan extension. Corsix doesn't allow that to occur - its generally not a high cost compared to the expenditure but its just another aspect of the game management that does occur.

Overdrafts operate the same way, calculated in original on end of month bank balance rather than a daily calculation. This means Corsix may sum smaller amounts across a month, more often and present an overdraft interest more often than the original, even though the amounts are generally lower. This also leads to overdrafts recording on end of month with positive bank balances.

The onEndDay method is also called before onEndMonth, which means the cost of wages and other monthly amounts (loan interest/heating etc), are not taken into account for any overdraft calculations until the end of 1st day the next month, which could be reduced or completely recovered by end of that day.

The original also penalises overdrafts compared to a loan by charging twice the interest than the loan rate rather than the +2% that currently occurs.

**Comment by lewri** [comment]
> Overdrafts operate the same way, calculated in original on end of month bank balance rather than a daily calculation. This means Corsix may sum smaller amounts across a month, more often and present an overdraft interest more often than the original, even though the amounts are generally lower. This also leads to overdrafts recording on end of month with positive bank balances.

Do you mean going negative incurred charges ontop of loans? Otherwise I'm not sure what overdraft means in this case.

**Reply by mugmuggy** [reply]
Yes, an overdraft (overdrawn) is negative bank balance. Ie have 0 dollars, take loan for 5k, spend 6k have 1k negative balance, 5k loan and 1k overdraft.

**Reply by ghost** [reply]
The original game rewards you for micro-managing at the end-of-the-month, so if you lower heating costs and pay off loans as much as possible, you reduce the costs. Then of course the next day, a new month you raise the heating again and take out loans for hiring new staff.

I can see why loans and heating are calculated daily if you view this as a bug in the original game,  just this kind of micro-management is good for new players who struggle in the early game.
